### PR TITLE
Improving MDN Javascript Fathead #268

### PR DESCRIPTION
**Pull request by bfmags**
**Follow-up from Improving MDN Javascript Fathead** https://github.com/duckduckgo/zeroclickinfo-fathead/pull/268

Make MDN JS perform faster with Wget
https://duck.co/ia/view/mdnjs

- [ ] MDN JS update parser
--- remove the forced lowercasing ---- https://github.com/duckduckgo/zeroclickinfo-fathead/issues/257

----------------------------------
IA Page: https://duck.co/ia/view/mdnjs
Forum topic: [Improve MDN JavaScript Fathead](https://forum.duckduckhack.com/t/improve-mdn-javascript-fathead/155)

This pull request is part of the [Programming Mission](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53): help us improve the results for [JavaScript related searches](https://forum.duckduckhack.com/t/javascript-search-overview/94)!

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@bfmags thanks for this, I'm not sure if it's a result of these changes, but I'm seeing duplicates again for lowecased names.

e.g an article for `length` but also a disambiguation as well as multiple redirects, all to different articles

Here's a picture of `git show` for this commit:

2__zeroclickinfo-spice__ssh_bastion__gcat__and_any_do

It looks like the trailing tabs have also been removed?

File: lib/fathead/mdnjs/parse.py [hunk 1]
```diff
@@ -248,30 +248,34 @@ def writerows(self):
...
-                        # check if not an Article
-                        if all(x in [keyword, strip_title] for x in self._writer.articles_index):
-                          return
+                        # skips redirect if already an article
+                        if strip_title.lower() not in self._writer.articles_index:
+                            self._writer.writerow({
+                              'title': strip_title,
+                              'type': 'R',
+                              'redirect': mdn.title
+                            })
```
Comment (moollaza) [inline lib/fathead/mdnjs/parse.py]: I think you need to add `strip_title.lower()` to the `articles_index` after this to prevent duplicates?

**Comment by moollaza** [comment]
Here's more example's of duplicates:
2__zeroclickinfo-spice__ssh_bastion__gcat__and_any_do

**Comment by bfmags** [comment]
@moollaza Thnx, made some changes that should fix those

File: lib/fathead/mdnjs/parse.py (hunk 1 above)
Comment (bfmags) [inline lib/fathead/mdnjs/parse.py]: should be fixed here 
https://github.com/duckduckgo/zeroclickinfo-fathead/pull/302/commits/a7619ededc157795ea9c40f8da16f5cfe076dcea

**Comment by MariagraziaAlastra** [comment]
It's on beta!

Fixed:
- [mdn document](https://beta.duckduckgo.com/?q=mdn+document&t=hd&ia=about)
- [mdn window document](https://beta.duckduckgo.com/?q=mdn+window+document&t=hd&ia=about)
- [mdn history](https://beta.duckduckgo.com/?q=mdn+window+history&t=hd&ia=about)
- [mdn close](https://beta.duckduckgo.com/?q=mdn+close&t=hd&ia=about)

Needs to be fixed:
- [javascript array sort](https://beta.duckduckgo.com/?q=javascript+array+sort&t=hd&ia=iatesting)

Thanks @bfmags :smile:

**Comment by MariagraziaAlastra** [comment]
@bfmags Latest is on beta again! [javascript array sort](https://beta.duckduckgo.com/?q=js+array+sort&ia=about) triggers now :smile:

Going to test this more and I'll let you know! Thanks a lot!

**Comment by MariagraziaAlastra** [comment]
@bfmags this LGTM, merging! :+1: 

Thanks a lot, I'll let you know when it's live :smile:

**Comment by MariagraziaAlastra** [comment]
@bfmags it's live!

- [js boolean](https://duckduckgo.com/?q=js+boolean&t=ffab&ia=about)
- [mdn window document](https://duckduckgo.com/?q=js+window+document&t=ffab&ia=about)
- [js array sort](https://duckduckgo.com/?q=js+array+sort&t=ffab&ia=about)

Great job! :smile:

**Pull request merged**